packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -86,4 +141,133 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
             order_id: orderID,
         });
     }
+
+    private async _onHostedMethodApprove(data: ApproveDataOptions, actions: ApproveActions) {
+        try {
+            const orderDetails = await actions.order.get();
+            const consignments = this._store.getState().consignments.getConsignmentsOrThrow();
+            const lineItems = this._getLineItems();
+            if (!this._paymentMethod?.id) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+            const methodId = this._paymentMethod?.id;
+            const shippingAddress = {
+                ...this._currentShippingAddress,
+                firstName: orderDetails.payer.name.given_name,
+                lastName: orderDetails.payer.name.surname,
+                email: orderDetails.payer.email_address,
+                address1: orderDetails.purchase_units[0].shipping.address.address_line_1,
+                address2: '',
+                stateOrProvince: '',
+                stateOrProvinceCode: '',
+                city: this._currentShippingAddress?.city || '',
+                countryCode: this._currentShippingAddress?.countryCode || '',
+                postalCode: this._currentShippingAddress?.postalCode || '',
```

Comment:
It might be better to set default values for `city`, `countryCode` and `postalCode` in `_transformAddress` method, to avoid using `this._currentShippingAddress` after `...this._currentShippingAddress`.

Reply:
Default values will not work, because this. _currentShippingAddress is optional, and ts still giving an error because city, countryCode and postalCode can be undefined because of this. _currentShippingAddress.

Reply:
We found another solution how to make it work